From a5c94848446c654d6c0da73a016e538f50ae554c Mon Sep 17 00:00:00 2001
From: Radu Bulie <radu.bulie@freescale.com>
Date: Mon, 25 Feb 2013 15:35:45 +0000
Subject: [PATCH 486/518] fmd: Copy specific parameters for enabling storage
 profile on KG schemes

Override storage profile flag and storage profile Id were not
copied in 32bit/64bit compat mode. This patch fixes the issue
and adds these two missing parameters.

Signed-off-by: Radu Bulie <radu.bulie@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 .../src/wrapper/lnxwrp_ioctls_fm_compat.c          |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/NetCommSw/src/wrapper/lnxwrp_ioctls_fm_compat.c b/drivers/net/ethernet/freescale/dpa/NetCommSw/src/wrapper/lnxwrp_ioctls_fm_compat.c
index c4c6a35..04ee23b 100644
--- a/drivers/net/ethernet/freescale/dpa/NetCommSw/src/wrapper/lnxwrp_ioctls_fm_compat.c
+++ b/drivers/net/ethernet/freescale/dpa/NetCommSw/src/wrapper/lnxwrp_ioctls_fm_compat.c
@@ -640,6 +640,9 @@ void compat_copy_fm_pcd_kg_scheme(
                sizeof(ioc_fm_pcd_kg_key_extract_and_hash_params_t));
         param->bypass_fqid_generation = compat_param->bypass_fqid_generation;
         param->base_fqid = compat_param->base_fqid;
+	param->override_storage_profile =
+					 compat_param->override_storage_profile;
+	param->storage_profile = compat_param->storage_profile;
         param->num_of_used_extracted_ors = compat_param->num_of_used_extracted_ors;
         memcpy(param->extracted_ors,
                compat_param->extracted_ors,
@@ -688,6 +691,9 @@ void compat_copy_fm_pcd_kg_scheme(
         memcpy(&compat_param->key_extract_and_hash_params, &param->key_extract_and_hash_params, sizeof(ioc_fm_pcd_kg_key_extract_and_hash_params_t));
         compat_param->bypass_fqid_generation = param->bypass_fqid_generation;
         compat_param->base_fqid = param->base_fqid;
+	compat_param->override_storage_profile =
+					       param->override_storage_profile;
+	compat_param->storage_profile =  param->storage_profile;
         compat_param->num_of_used_extracted_ors = param->num_of_used_extracted_ors;
         memcpy(compat_param->extracted_ors, param->extracted_ors, IOC_FM_PCD_KG_NUM_OF_GENERIC_REGS * sizeof(ioc_fm_pcd_kg_extracted_or_params_t));
         compat_param->next_engine = param->next_engine;
-- 
1.7.5.4

